planner/core/optimizer.go

```diff
@@ -102,7 +103,7 @@ func DoOptimize(flag uint64, logic LogicalPlan) (PhysicalPlan, error) {
 	if err != nil {
 		return nil, err
 	}
-	if !AllowCartesianProduct && existsCartesianProduct(logic) {
+	if !AllowCartesianProduct.Load() && existsCartesianProduct(logic) {
```

Comment:
how many ns does it take to perform a `AllowCartesianProduct.Load()` operation?

Reply:
`uber.Atomic` just a simple wrapper around `golang.Atomic`.
And here is the benchmark of `uber.Atomic` and `golang.Atomic`:
```
BenchmarkUberAtomicFloat64-4    200000000                8.82 ns/op
BenchmarkUberAtomicInt64-4      200000000                8.57 ns/op
BenchmarkUberAtomicUint64-4     200000000                8.57 ns/op
BenchmarkUberAtomicBool-4       200000000                9.38 ns/op
BenchmarkGoAtomicInt64-4        200000000                8.65 ns/op
BenchmarkGoAtomicUnt64-4        200000000                8.67 ns/op
```

Reply:
can we make the whole config as an atomic value, only replace the config when it's reloaded?

Reply:
Now the whole config variable is already a atomic value(atomic pointer) and we replace it when reloading by `c.Store(newConf)`.

Reply:
but this operation seems a little expensive, do we have another lock free method?

Reply:
emm, but there is no lock on the whole config variable, it's already a lock free method(atomic Store/Load).